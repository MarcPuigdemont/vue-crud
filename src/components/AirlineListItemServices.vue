<template>
  <b-row>
    <b-col v-for="service in services" :key="service.name">
      <i class="material-icons" :title="service.name" :disabled="!service.available">{{ getIcon(service.name) }}</i>
    </b-col>
  </b-row>
</template>
<script>
const icons = {
  bags: 'work',
  checkin: 'where_to_vote',
  seats: 'airline_seat_recline_normal',
};
export default {
  props: { services: Array },
  methods: {
    getIcon: serviceName => icons[serviceName] || 'error',
  },
};
</script>
<style scoped>
i[disabled] {
  color: rgba(0, 0, 0, 0.1);
}
</style>
