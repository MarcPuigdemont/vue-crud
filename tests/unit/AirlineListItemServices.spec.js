import { mount } from '@vue/test-utils';
import AirlineListItemServices from '@/components/AirlineListItemServices.vue';

describe('AirlineListItemServices', () => {
  test('should match the snapshot', () => {
    const wrapper = mount(AirlineListItemServices, { propsData: { services: [{ name: 'bags' }, { name: 'checkin' }, { name: 'seats' }] } });
    expect(wrapper.element).toMatchSnapshot();
  });
  test('should return error icon if service name is not correct', () => {
    const wrapper = mount(AirlineListItemServices, { propsData: { services: [{ name: 'not_bags' }] } });
    expect(wrapper.element).toMatchSnapshot();
  });
});
