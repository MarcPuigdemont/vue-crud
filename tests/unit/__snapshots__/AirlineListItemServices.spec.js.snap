// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirlineListItemServices should match the snapshot 1`] = `
<div
  class="row"
>
  <div
    class="col"
  >
    <i
      class="material-icons"
      data-test="list-row-service-bags"
      disabled="disabled"
      title="bags"
    >
      
      work
    
    </i>
  </div>
  <div
    class="col"
  >
    <i
      class="material-icons"
      data-test="list-row-service-checkin"
      disabled="disabled"
      title="checkin"
    >
      
      where_to_vote
    
    </i>
  </div>
  <div
    class="col"
  >
    <i
      class="material-icons"
      data-test="list-row-service-seats"
      disabled="disabled"
      title="seats"
    >
      
      airline_seat_recline_normal
    
    </i>
  </div>
</div>
`;

exports[`AirlineListItemServices should return error icon if service name is not correct 1`] = `
<div
  class="row"
>
  <div
    class="col"
  >
    <i
      class="material-icons"
      data-test="list-row-service-not_bags"
      disabled="disabled"
      title="not_bags"
    >
      
      error
    
    </i>
  </div>
</div>
`;
